### PR TITLE
A few more OpenShift tips

### PR DESCRIPTION
**Pull request by bianchi2**
A few additions to OpenShift gotchas, namely explanation of the nature of permission issues when generating config files from Jinja templates and how to fix it without attaching anyuid SCC by building a custom image from an official one.

### CODE REVIEW

File: docs/docs/platforms/OPENSHIFT.md [hunk 1]
```diff
@@ -28,11 +34,11 @@ It depends on the storage backend though.
 
 As an alternative, (if letting containers run as pre-defined users is not possible), set `product_name.securityContextEnabled` to `false`, for example, `confluence.securityContextEnabled: false`
 As a result the container will start as a user with an OpenShift generated ID.
-Typically, NFS permission fixer job isn't required when no security context is set.
+You will also need to disable NFX permission fixer init container as it start as roo. Set `volumes.sharedHome.nfsPermissionFixer.enabled=false` to false.
```
Comment (nghazali) [inline docs/docs/platforms/OPENSHIFT.md:37]: `it starts as root`?

File: docs/docs/platforms/OPENSHIFT.md [hunk 2]
```diff
@@ -47,6 +53,31 @@ volumes:
       emptyDir: {}
 ```
 
+While it's possible to declare runtime volumes for empty directories, it is not possible for `${APPLICATION_HOME}/conf`. When starting up, Jira and Confluence generate a few configuration files which is a part of image entrypoint. Without anyuid SCC, an unprivileged user can't write to `${APPLICATION_HOME}/conf`. When starting Jira in OpenShift without anyuid SCC attached to jira service account, you will see the following log:
```
Comment (nghazali) [inline docs/docs/platforms/OPENSHIFT.md:56]: `Without any uid SCC`?

File: docs/docs/platforms/OPENSHIFT.md [hunk 3]
```diff
@@ -47,6 +53,31 @@ volumes:
...
+WARNING:root:Permission problem writing '/opt/atlassian/jira/conf/server.xml'; skipping
+INFO:root:Generating /opt/atlassian/jira/atlassian-jira/WEB-INF/classes/seraph-config.xml from template seraph-config.xml.j2
+WARNING:root:Permission problem writing '/opt/atlassian/jira/atlassian-jira/WEB-INF/classes/seraph-config.xml'; skipping
+INFO:root:Generating /var/atlassian/application-data/jira/dbconfig.xml from template dbconfig.xml.j2
+WARNING:root:Could not chown path /var/atlassian/application-data/jira/dbconfig.xml to jira:jira due to insufficient permissions.
+INFO:root:Running Jira with command '/opt/atlassian/jira/bin/start-jira.sh', arguments ['/opt/atlassian/jira/bin/start-jira.sh', '-fg']
+executing as current user
+```
+
+While this is a non fatal error and Jira is able to proceed with startup, failure to properly generate configuration files like server.xml will result in a number of error when using Jira.
```
Comment (nghazali) [inline docs/docs/platforms/OPENSHIFT.md:71]:
```suggestion
While this is a non-fatal error and Jira is able to proceed with startup, failure to properly generate configuration files like server.xml will result in a number of errors when using Jira.
```

File: docs/docs/platforms/OPENSHIFT.md (hunk 1 above)
Comment (tarka) [inline docs/docs/platforms/OPENSHIFT.md:37]: NFX should be NFS?

File: docs/docs/platforms/OPENSHIFT.md (hunk 2 above)
Comment (tarka) [inline docs/docs/platforms/OPENSHIFT.md:56]: "of image entrypoint" -> "of the image entrypoint"?

File: docs/docs/platforms/OPENSHIFT.md [hunk 4]
```diff
@@ -47,6 +53,31 @@ volumes:
...
+
+To mitigate the problem, either attach anyuid policy to jira (or confluence) service account **or** build your own container image if existing security practices do not allow anyuid. You need to inherit Jira (or Confluence) official image and make a few directories/files writable for users belonging to a `root` group (which users in OpenShfit containers belong to):
```
Comment (tarka) [inline docs/docs/platforms/OPENSHIFT.md:73]: "need to inherit Jira" -> "need to inherit the Jira"?

**Review by tarka — CHANGES_REQUESTED** [review]

File: docs/docs/platforms/OPENSHIFT.md (hunk 2 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md:56]: SCC is called `anyuid` https://docs.openshift.com/container-platform/4.6/authentication/managing-security-context-constraints.html#default-sccs_configuring-internal-oauth

File: docs/docs/platforms/OPENSHIFT.md (hunk 1 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: Fixed, thanks

File: docs/docs/platforms/OPENSHIFT.md (hunk 3 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: Fixed

File: docs/docs/platforms/OPENSHIFT.md (hunk 1 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: Fixed

File: docs/docs/platforms/OPENSHIFT.md (hunk 4 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: Fixed

File: docs/docs/platforms/OPENSHIFT.md (hunk 2 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: Fixed

Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: I have pushed a commit with `anyuid` font to make sure it's clear it's the scc name

File: docs/docs/platforms/OPENSHIFT.md [hunk 5]
```diff
@@ -35,11 +41,11 @@ It depends on the storage backend though.
 
 As an alternative, (if letting containers run as pre-defined users is not possible), set `product_name.securityContextEnabled` to `false`, for example, `confluence.securityContextEnabled: false`
 As a result the container will start as a user with an OpenShift generated ID.
-Typically, NFS permission fixer job isn't required when no security context is set.
+You will also need to disable NFS permission fixer init container as it start as root. Set `volumes.sharedHome.nfsPermissionFixer.enabled=false` to false.
```
Comment (tarka) [inline docs/docs/platforms/OPENSHIFT.md:44]: Minor rephrasing:
"You will also need to disable the NFS permission fixer init container as it starts as root. To do this set `volumes.sharedHome.nfsPermissionFixer.enabled=false` to false."

**Review by tarka — CHANGES_REQUESTED** [review]

File: docs/docs/platforms/OPENSHIFT.md (hunk 5 above)
Comment (bianchi2) [reply docs/docs/platforms/OPENSHIFT.md]: Done

**Review by tarka — DISMISSED** [review]

File: docs/docs/platforms/OPENSHIFT.md [hunk 6]
```diff
@@ -33,13 +39,13 @@ It depends on the storage backend though.
 
 ## Disable security context
 
-As an alternative, (if letting containers run as pre-defined users is not possible), set `product_name.securityContextEnabled` to `false`, for example, `confluence.securityContextEnabled: false`
+As an alternative, (if letting containers run as pre-defined users is not possible), set `product_name.securityContextEnabled` to `false`, for example, `confluence.securityContextEnabled: false`.
 As a result the container will start as a user with an OpenShift generated ID.
-Typically, NFS permission fixer job isn't required when no security context is set.
+You will also need to disable NFS permission fixer init container as it start as root. Set `volumes.sharedHome.nfsPermissionFixer.enabled` to false.
```
Comment (yzha645) [inline docs/docs/platforms/OPENSHIFT.md:44]:
```suggestion
You will also need to disable NFS permission fixer init container as it starts as root. Set `volumes.sharedHome.nfsPermissionFixer.enabled` to false.
```

File: docs/docs/platforms/OPENSHIFT.md [hunk 7]
```diff
@@ -54,6 +60,31 @@ volumes:
...
+WARNING:root:Permission problem writing '/opt/atlassian/jira/atlassian-jira/WEB-INF/classes/seraph-config.xml'; skipping
+INFO:root:Generating /var/atlassian/application-data/jira/dbconfig.xml from template dbconfig.xml.j2
+WARNING:root:Could not chown path /var/atlassian/application-data/jira/dbconfig.xml to jira:jira due to insufficient permissions.
+INFO:root:Running Jira with command '/opt/atlassian/jira/bin/start-jira.sh', arguments ['/opt/atlassian/jira/bin/start-jira.sh', '-fg']
+executing as current user
+```
+
+While this is a non-fatal error and Jira is able to proceed with startup, failure to properly generate configuration files like server.xml will result in a number of errors when using Jira.
+
+To mitigate the problem, either attach anyuid policy to jira (or confluence) service account **or** build your own container image if existing security practices do not allow anyuid. You need to inherit the Jira (or Confluence) official image and make a few directories/files writable for users belonging to a `root` group (which users in OpenShfit containers belong to):
```
Comment (yzha645) [inline docs/docs/platforms/OPENSHIFT.md:80]:
```suggestion
To mitigate the problem, either attach anyuid policy to Jira (or Confluence) service account **or** build your own container image if existing security practices do not allow anyuid. You need to inherit the Jira (or Confluence) official image and make a few directories/files writable for users belonging to a `root` group (which users in OpenShfit containers belong to):
```

**Review by yzha645 — APPROVED** [review]

**Review by yzha645 — APPROVED** [review]

**Review by nghazali — APPROVED** [review]

**Pull request merged**